src/state/swap/hooks.tsx

```diff
@@ -71,7 +72,10 @@ const BAD_RECIPIENT_ADDRESSES: { [address: string]: true } = {
 }
 
 // from the current swap inputs, compute the best trade and return it.
-export function useDerivedSwapInfo(state: SwapState): {
+export function useDerivedSwapInfo(
+  state: SwapState,
+  defaultChainId?: SupportedChainId
```

Comment:
nit: I would rephrase this to just `chainId`, because we want to get information for a given chain and there's no default chain anymore. I would rather let Swap page (main page) pass explicit `chainId` from web3react explicitly rather than handle that inside. And let's make it a required argument.

Reply:
Then I would remove `chainId = useWeb3React` from `[src/hooks/Tokens.ts](https://github.com/Uniswap/interface/pull/6332/files/8ddb100fbc639d4191edd6137c9a87e648e89595#diff-3340e3e31b3f7eff7bb21323f4ee0e638c6f05d468225e0b46511ceee4462d8e)` and explicitly pass it on Swap page too.